sklearn/tests/test_cross_validation.py

```diff
@@ -362,13 +362,48 @@ def test_shuffle_split():
     ss3 = cval.ShuffleSplit(10, test_size=np.int32(2), random_state=0)
     for typ in six.integer_types:
         ss4 = cval.ShuffleSplit(10, test_size=typ(2), random_state=0)
-    for t1, t2, t3, t4 in zip(ss1, ss2, ss3, ss4):
```

Comment:
This test is painful to read. I would suggest putting the ShuffleSplits in lists and do a for loop over the indices. WDYT?